From: Jo Zzsi <jozzsicsataban@gmail.com>
Date: Wed, 13 Nov 2024 06:44:39 -0500
Subject: fix(dracut-init): add compatibility with Debian/Ubuntu for libdirs
 detection

Make sure if /lib/ is included in libdirs when needed.

This change is required for multipath dracut module to work properly
on Debian/Ubuntu based distributions.

Origin: upstream, https://github.com/dracut-ng/dracut-ng/pull/950
---
 dracut-init.sh                    | 5 ++++-
 test/TEST-04-FULL-SYSTEMD/test.sh | 2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/dracut-init.sh b/dracut-init.sh
index 1ce0d7e..11ae3d1 100755
--- a/dracut-init.sh
+++ b/dracut-init.sh
@@ -91,7 +91,10 @@ if ! [[ $libdirs ]]; then
         && [[ -d $dracutsysrootdir/lib64 ]]; then
         libdirs+=" /lib64"
         [[ -d $dracutsysrootdir/usr/lib64 ]] && libdirs+=" /usr/lib64"
-    else
+
+    fi
+
+    if [[ -d $dracutsysrootdir/lib ]]; then
         libdirs+=" /lib"
         [[ -d $dracutsysrootdir/usr/lib ]] && libdirs+=" /usr/lib"
     fi
diff --git a/test/TEST-04-FULL-SYSTEMD/test.sh b/test/TEST-04-FULL-SYSTEMD/test.sh
index c2bff28..8476677 100755
--- a/test/TEST-04-FULL-SYSTEMD/test.sh
+++ b/test/TEST-04-FULL-SYSTEMD/test.sh
@@ -128,7 +128,7 @@ EOF
     echo -n test > /tmp/key
 
     test_dracut \
-        -m "dracut-systemd systemd-ac-power systemd-coredump systemd-creds systemd-cryptsetup systemd-integritysetup systemd-ldconfig systemd-pcrphase systemd-pstore systemd-repart systemd-sysext systemd-veritysetup" \
+        -a "dracut-systemd systemd-ac-power systemd-coredump systemd-creds systemd-cryptsetup systemd-integritysetup systemd-ldconfig systemd-pcrphase systemd-pstore systemd-repart systemd-sysext systemd-veritysetup" \
         --add-drivers "btrfs" \
         -i "/tmp/key" "/etc/key" \
         "$TESTDIR"/initramfs.testing
